File: CSharp/src/Sorts/CountSort.cs
namespace Algorithm.Sorts;

public class CountSort
{
    public int[] Sort(int[] array)
    {
        var b = new int[array.Length];
        var c = new int[array.Length];

        for (var i = 0; i < array.Length; i++)
        {
            c[array[i]] += 1;
        }

        for (var i = 1; i < b.Length; i++)
        {
            c[i] += c[i - 1];
        }

        for (var i = array.Length - 1; i >= 0; i--)
        {
            b[c[array[i]] - 1] = array[i];
            c[array[i]] -= 1;
        }

        return b;
    }
}

File: Python/src/sorts/count_sort.py
def count_sort(array):
    b = [0] * len(array)
    c = [0] * len(array)

    for i in range(0, len(array)):
        c[array[i]] += 1

    for i in range(1, len(c)):
        c[i] += c[i - 1]

    for i in range(len(array) - 1, -1, -1):
        b[c[array[i]] - 1] = array[i]
        c[array[i]] -= 1
    
    return b